pkg/apis/autoscaling/v1alpha1/pa_validation.go

```diff
@@ -62,11 +61,7 @@ func (pa *PodAutoscaler) validateMetric() *apis.FieldError {
 			// Leave other classes of PodAutoscaler alone.
 			return nil
 		}
-		return &apis.FieldError{
-			Message: fmt.Sprintf("Unsupported metric %q for PodAutoscaler class %q",
-				metric, pa.Class()),
-			Paths: []string{"annotations[autoscaling.knative.dev/metric]"},
-		}
+		return apis.ErrInvalidValue(metric, "annotations[autoscaling.knative.dev/metric]")
```

Comment:
I'd go with `apis.ErrInvalidValue(metric, "autoscaling.knative.dev/metric").Via("annotations")`.
I think that's how most of the `apis/autoscaling/annotation_validation.go` works.
Also, which reminds me, may be we should just move the validation there (unless it already is) so the object manipulation fails at webhook level and here presume validated so just return.

Reply:
Changed to `apis.ErrInvalidValue(metric, "autoscaling.knative.dev/metric").ViaField("annotations")`.

The validations in `apis/autoscaling/annotation_validation.go` don't depend on `PA`. However `autoscaling.knative.dev/metric` depends on which `PA` type is used.